sdk/metric/internal/sum.go

```diff
@@ -17,15 +17,28 @@
 
 package internal // import "go.opentelemetry.io/otel/sdk/metric/internal"
 
-import "go.opentelemetry.io/otel/attribute"
+import (
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+)
 
 // sum summarizes a set of measurements as their arithmetic sum.
 type sum[N int64 | float64] struct {
-	// TODO(#2972): implement.
+	sync.Mutex
+
+	values map[attribute.Set]N
+}
```

Comment:
Out of curiosity, is this something listed in the specification?

> limit the memory usage of the cumulative

I was unclear if it was allowed or recommended anywhere.  It it is, I would like to track it as work needed for the beta release.

Reply:
There are several salient spec-level questions surrounding this:

1. Whether an SDK should/must take efforts to limit memory growth when synchronous instruments are used with high cardinality and **Delta** temporality.  It's tricky to get this right -- that's where a lot of the complexity in the old SDK comes from -- so I view it as an implementation detail. If a user is coming to OTel from any modern statsd ecosystem (e.g., DataDog), they will have been using a library that **is expected to limit memory growth in exactly this scenario**. For this reason I view it as a Very Important implementation detail and would be glad to see it written in the specification.
2. Whether _and precisely how_ should/must an SDK limit memory growth for cumulative temporality. This is a relatively large grey area that requires at least following both the Prometheus guidance (don't reset very often) and the OTel guidance (reset your start time); probably should also include definite use of a staleness marker. When enough people care, this can be specified. However note that it is somewhat difficult to come up with the "right" answer for the start time.

Reply:
I'll add a todo and an issue to track the expected memory bounding work (and numeric limit handling) so we can address these in the next release cycle.

Reply:
Added: https://github.com/open-telemetry/opentelemetry-go/pull/3000/commits/45b07edd5bca7abe997430535ac5e96ff976fb6f#diff-12acd31cc033269be317c16835d2f75da31d39f97a1430c2e1e67c723628c14eR109-R112